benches/bench.rs

```diff
@@ -24,20 +24,22 @@ fn simple() -> String {
             _ => continue,
         }
     }"#;
-    let message = Level::Error.title("mismatched types").id("E0308").snippet(
-        Snippet::source(source)
-            .line_start(51)
-            .origin("src/format.rs")
-            .annotation(
-                Level::Warning
-                    .span(5..19)
-                    .label("expected `Option<String>` because of return type"),
-            )
-            .annotation(
-                Level::Error
-                    .span(26..724)
-                    .label("expected enum `std::option::Option`"),
-            ),
+    let message = Level::Error.message("mismatched types").id("E0308").group(
+        Group::new().element(
```

Comment:
Could you explain the new API in the commit or PR?

For example, why was a `Group` container chosen over `group` accepting a `Vec` or `IntoIterator`